mne_bids/read.py

```diff
@@ -415,17 +426,31 @@ def read_raw_bids(bids_fname, bids_root, extra_params=None,
             coordsystem_json = json.load(fin)
 
         # Get coordinate frames that electrode coordinates are in
+        # Note: all coordinate frames should be lower-case
         if kind == "meg":
             coord_frame = coordsystem_json['MEGCoordinateSystem']
+            coord_unit = coordsystem_json['MEGCoordinateUnits']
         elif kind == "ieeg":
             coord_frame = coordsystem_json['iEEGCoordinateSystem']
-        else:
+            coord_unit = coordsystem_json['iEEGCoordinateUnits']
+        else:  # noqa
+            # XXX should add support of coordsystem.json for EEG
             raise RuntimeError("Kind {} not supported yet for "
                                "coordsystem.json and "
                                "electrodes.tsv.".format(kind))
+
+        coord_frame = coord_frame.lower()  # MNE uses lower case
+        if kind == 'ieeg':
+            # XXX: improve reading with a `_write_dig_bids`
```

Comment:
reading with a `_write_dig_bids`?

Reply:
means that this code block should be moved into a dedicated private function _write_dig_bids

Reply:
> reading with a `_write_dig_bids`?

I'm happy to try to help out w/ this later (maybe next week), but for the purposes of this PR, I just want to solve the "scaling" in units and the arbitrary coordinate systems for iEEG.